synapse/handlers/register.py

```diff
@@ -465,6 +492,13 @@ async def _join_rooms(self, user_id: str) -> None:
                         join_rule = join_rules_event.content.get("join_rule", None)
                         requires_invite = join_rule and join_rule != JoinRules.PUBLIC
 
+                        if requires_invite and self.server_name not in hosts_in_room:
+                            raise SynapseError(
```

Comment:
We probably don't want to raise in the middle of registration.

Reply:
It's already inside a try/except.